Marlin/Marlin_main.cpp

```diff
@@ -6964,6 +7032,7 @@ void disable_all_steppers() {
 void idle() {
   manage_heater();
   manage_inactivity();
+  host_keepalive();
```

Comment:
It's subtle, but `idle2()` passes `true` to `manage_inactivity`, whereas `idle()` passes no argument. The only difference is that with `idle2` (I will try a more descriptive name) the steppers will never be disabled (according to the `M18`/`M84` `stepper_inactive_time` value, `DEFAULT_STEPPER_DEACTIVE_TIME`, and `DISABLE_INACTIVE_[XYZE]`). Actually, the only reason `idle2()` exists is because this is what `gcode_M600` was doing instead of calling `idle()` – so `idle2()` really just there to preserve existing behavior.